code/go/0chain.net/smartcontract/storagesc/collect_reward.go

```diff
@@ -0,0 +1,57 @@
+package storagesc
+
+import (
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+	"0chain.net/smartcontract/stakepool"
+)
+
+func (ssc *StorageSmartContract) collectReward(
+	txn *transaction.Transaction,
+	input []byte,
+	balances cstate.StateContextI,
+) (string, error) {
+
+	var prr stakepool.CollectRewardRequest
+	if err := prr.Decode(input); err != nil {
+		return "", common.NewErrorf("pay_reward_failed",
+			"can't decode request: %v", err)
+	}
+
+	usp, err := stakepool.GetUserStakePool(prr.ProviderType, txn.ClientID, balances)
+	if err != nil {
+		return "", common.NewErrorf("pay_reward_failed",
+			"can't get related user stake pools: %v", err)
+	}
+
+	providerId := usp.Find(prr.PoolId)
+	if len(providerId) == 0 {
+		return "", common.NewErrorf("pay_reward_failed",
+			"user %v does not own stake pool %v", txn.ClientID, prr.PoolId)
+	}
+
+	sp, err := ssc.getStakePool(providerId, balances)
+	if err != nil {
+		return "", common.NewErrorf("pay_reward_failed",
+			"can't get related stake pool: %v", err)
+	}
+
+	_, err = sp.MintRewards(
+		txn.ClientID, prr.PoolId, providerId, prr.ProviderType, usp, balances)
+	if err != nil {
+		return "", common.NewErrorf("pay_reward_failed",
+			"error emptying account, %v", err)
+	}
```

Comment:
Use the stakepool.CollectRewards() perhaps, seems they do the same thing.

Reply:
The getStakePool and Save functions are different. I have removed `stakepool.CollectRewards()`, It can be put back if it seems necessary for miner, sharder and authoriser stake pools, but is not needed for blobbers and validators.